File: homework/08HW/10.py
#統計學生總分與平均
#輸入一個學生資料的 list，每筆資料是字典，寫一個函數 summary(data)，印出每位學生的總分與平均。

#範例輸入：

#students = [
#    {'name': 'Alice', 'scores': [90, 80, 70]},
#    {'name': 'Bob', 'scores': [100, 85, 95]}
#]

def summary(data):
    for student in data:
        total = sum(student['scores'])
        average = total / len(student['scores']) if student['scores'] else 0
        print(f"{student['name']} 總分: {total}, 平均: {average:.1f}")

students = [
    {'name': 'Alice', 'scores': [90, 80, 70]},
    {'name': 'Bob', 'scores': [100, 85, 95]}
]
summary(students)

File: homework/03javascrip練習/10.js
/*
物件遍歷

建立一個物件 scores = { Alice: 85, Bob: 92, Charlie: 78 }。
使用 for...in 迴圈遍歷 scores 物件，輸出 "Alice 的分數是 85" 這樣的資訊。
*/

let scores = { Alice: 85, Bob: 92, Charlie:78}
let text="";
for(let x in scores.Alice){
    text+= scores[x];
    console.log("Alice 的分數是",(text));
}

/*gpt's version
let scores = { Alice: 85, Bob: 92, Charlie: 78 };

for (let name in scores) {  // 遍歷物件的屬性
    console.log(`${name} 的分數是 ${scores[name]}`);
}
*/

